### ISSUE TITLE
Strip comments from requirement lines

### ISSUE THREAD

**Issue by hugorodgerbrown** [yunojuno/django-package-monitor#17]
Pip supports inline comments in the requirements file:

```
social-auth-core==1.3.0   # via social-auth-app-django
sqlparse==0.2.3           # via django-request-token
```

As we parse the file into individual requirements, we should strip out any such comments. We can't split purely on `#` as this is used for appending package names to pinned commits, e.g.:

```
-e git+git://github.com/jsocol/django-waffle.git@a2c41676e4f9be54dab2522b10dec09189245e74#egg=django_waffle
```

Proposal is to split on ' #'.

**Comment by hugorodgerbrown** [comment]
`line.split(' #')[0].strip()`

**Comment by hugorodgerbrown** [comment]
Fixed by #16
